Add Jenkins pipeline support.

sudswas opened the pull request:
This commit adds the support for Jenkins Kubernetes Pipeline.
This adds the Jenkinsfile, that has configurations specific to
ICP to make a CI/CD workflow for this application.

Some values are hardcoded for with local URLs, that should be
replaced while running on a ICP cluster that as a different cluster
name.

Commit:
Add Jenkins pipeline support.

This commit adds the support for Jenkins Kubernetes Pipeline.
This adds the Jenkinsfile, that has configurations specific to
ICP to make a CI/CD workflow for this application.

Some values are hardcoded for with local URLs, that should be
replaced while running on a ICP cluster that as a different cluster
name.

Commit:
Add helm chart

This commit adds the Helm chart for the flaskapp.
Once this repo is cloned, one should be simply able to run
helm install to deploy this application on a kubernetes cluster.

sudswas committed:
Update to the README

Changes to the README file

sudswas committed:
Merge pull request #1 from sudswas/sudswas-patch-1

Update to the README

Commit:
Test a dummy commit

This is a dummy commit for the demo.

bpradipt commented on Jenkinsfile:
```diff
@@ -0,0 +1,25 @@
+    docker.withRegistry('https://mycluster.icp:8500/', 'docker'){
+    stage "Build"
+
+        def pcImg = docker.build("mycluster.icp:8500/default/flask-app:${env.BUILD_ID}", "-f Dockerfile.ppc64le .")
```
Instead of hardcoding the registry server, can you set the values in env variable and reference it here.
eg.
env.REGISTRY=mycluster.icp:8500
env.REGISTRY_NAMESPACE=default

sudswas committed:
Merge pull request #2 from sudswas/helm

Helm

Commit:
Added sample jenkins file

This commit adds a sample jenkinsfile obtained from:
https://github.com/kubernetes/charts/blob/master/stable/jenkins/values.yaml

and then modified with values for Power images.

sudswas committed:
Merge pull request #3 from sudswas/jenkins-add

Added sample jenkins file

sudswas committed:
Update README.md

Commit:
Changes to be shown for demo
